src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -127,93 +180,163 @@ def _set_aladdin_recommendations(self):
+            decorated_description = [(Style.SECONDARY, recommendation['description'] + '\n')]
+            decorated_recommendations.append((decorated_command, decorated_description))
+
+        # add reference link as a recommendation
+        decorated_link = [(Style.HYPERLINK, OVERVIEW_REFERENCE)]
+        if self.aladdin_recommendations:
+            decorated_link = [(Style.HYPERLINK, self.aladdin_recommendations[0]['link'])]
+
+        decorated_description = [(Style.SECONDARY, 'Read more about the command in reference docs')]
+        decorated_recommendations.append((decorated_link, decorated_description))
 
-            # sort the candidates by the number of matched arguments and total arguments
-            candidates = get_sorted_candidates(target_args, candidate_args_list)
-            if candidates:
-                index = candidates[0]['index']
-                recommend_command = candidate_commands[index]
-
-        # fallback to use the first recommended command from Aladdin
-        elif self.aladdin_recommendations:
-            recommend_command = self.aladdin_recommendations[0]
         # set the recommend command into Telemetry
-        self._set_recommended_command_to_telemetry(recommend_command)
-        # replace the parameter values
-        recommend_command = self._replace_parameter_values(recommend_command)
+        self._set_recommended_command_to_telemetry(raw_commands)
+
+        return decorated_recommendations
 
-        return recommend_command
+    def _set_recommended_command_to_telemetry(self, raw_commands):
+        """Set the recommended commands to Telemetry
 
-    def _set_recommended_command_to_telemetry(self, recommend_command):
-        """Set the recommended command to Telemetry for analysis. """
+        Aladdin recommended commands and commands from CLI help examples are
+        set to different properties in Telemetry.
+
+        :param raw_commands: The recommended raw commands
+        :type raw_commands: list
+        """
 
-        if recommend_command in self.aladdin_recommendations:
-            telemetry.set_debug_info('AladdinRecommendCommand', recommend_command)
-        elif recommend_command:
-            telemetry.set_debug_info('ExampleRecommendCommand', recommend_command)
+        if self.aladdin_recommendations:
+            telemetry.set_debug_info('AladdinRecommendCommand', ';'.join(raw_commands))
+        else:
+            telemetry.set_debug_info('ExampleRecommendCommand', ';'.join(raw_commands))
 
     def _disable_aladdin_service(self):
         """Decide whether to disable aladdin request when a command fails.
+
         The possible cases to disable it are:
-            1. In air-gapped clouds
-            2. In testing environments
+            1. CLI context is missing
+            2. In air-gapped clouds
+            3. In testing environments
+
+        :return: whether Aladdin service need to be disabled or not
+        :type: bool
```

Comment:
in 2,3, we could still provide recommendation from help right? Seems need to separate it in command_recommender

Reply:
They have already been separated, we could provide recommendations from help examples even though Aladdin service is disabled.